### PR TITLE
Feat: rebase sequencer on blockifier fork

### PR DESCRIPTION
**Pull request by bajpai244**
# Pull Request type

This pr enables our sequencer to work on top of our own blockifier and cairo-vm fork.

Resolves: #546 

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## What is the new behavior?

The codebase has been rebased on top of our own blockifier and cairo-vm fork.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by ClementWalter — DISMISSED** [review]
So this is just to properly pin version?

**Comment by bajpai244** [comment]
> So this is just to properly pin version?

@ClementWalter yes you are right.

**Review by greged93 — CHANGES_REQUESTED** [review]

File: crates/ef-testing/src/evm_sequencer/mod.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ impl KakarotSequencer {
         for (k, v) in storage {
             (&mut self.0.state).set_storage_at(
                 *KAKAROT_ADDRESS,
-                get_storage_var_address(k, &[]).unwrap(), // safe unwrap: all vars are ASCII
+                get_storage_var_address(k, &[]), // safe unwrap: all vars are ASCII
```
[the remark this answers is not in the document]
Comment (bajpai244) [reply crates/ef-testing/src/evm_sequencer/mod.rs]: resolve with [this](https://github.com/kkrt-labs/ef-tests/pull/549/commits/a61a8c5fa9d629480c85387818470cdd96501462) commit.

File: crates/ef-testing/src/evm_sequencer/evm_state.rs [hunk 2]
```diff
@@ -109,7 +109,7 @@ impl EvmState for KakarotSequencer {
         for ((var, keys), v) in storage {
             (&mut self.0.state).set_storage_at(
                 starknet_address,
-                get_storage_var_address(var, &keys).unwrap(), // safe unwrap: all vars are ASCII
```
[the remark this answers is not in the document]
Comment (bajpai244) [reply crates/ef-testing/src/evm_sequencer/evm_state.rs:112]: resolve with [this](https://github.com/kkrt-labs/ef-tests/pull/549/commits/a61a8c5fa9d629480c85387818470cdd96501462) commit.

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**